replication-api-impl/src/main/java/com/connexta/replication/api/impl/data/ReplicationItemManagerImpl.java

```diff
@@ -61,17 +67,33 @@ public ReplicationItemManagerImpl(ItemRepository itemRepository, SolrTemplate so
 
   @Override
   public ReplicationItem get(String id) {
-    return itemRepository
-        .findById(id)
-        .map(ReplicationItemImpl::new)
-        .orElseThrow(NotFoundException::new);
+    try {
+      return itemRepository
+          .findById(id)
+          .map(ReplicationItemImpl::new)
+          .orElseThrow(NotFoundException::new);
+    } catch (NonTransientDataAccessException e) {
+      throw new NonTransientReplicationPersistenceException(e);
+    } catch (TransientDataAccessException e) {
+      throw new TransientReplicationPersistenceException(e);
+    } catch (RecoverableDataAccessException e) {
+      throw new RecoverableReplicationPersistenceException(e);
+    }
```

Comment:
✏️ I'm pretty sure you can make a method that takes a function and run the function in this try catch there so you don't have to repeat them in each method.

Reply:
The problem is that some of these methods don't return anything while others do. I opted for cut & paste since the class was simple enough to start with.